articles/114_generated_interfaces_python.md

```diff
@@ -0,0 +1,110 @@
+
+## Generated files
+
+Following the Python conventions the namespace hierarchy is mapped to a folder structure.
+The package and module names and therefore the folder and file names use lowercase alphanumeric characters with underscores for separating words (following PEP 8).
+Python files end with `.py`.
+
+## Messages
+
+For a message a Python `class` with the same name is generated in the file `<my_message_name>.py`.
+
+### Types
+
+#### Mapping of primitive types
+
+| ROS type | Python type                  |
+| -------- | ---------------------------- |
+| bool     | builtins.bool                |
+| byte     | builtins.bytes with length 1 |
+| char     | builtins.str with length 1   |
+| float32  | builtins.float               |
+| float64  | builtins.float               |
+| int8     | builtins.int                 |
+| uint8    | builtins.int                 |
+| int16    | builtins.int                 |
+| uint16   | builtins.int                 |
+| int32    | builtins.int                 |
+| uint32   | builtins.int                 |
+| int64    | builtins.int                 |
+| uint64   | builtins.int                 |
+| string   | builtins.str                 |
+
+#### Mapping of arrays and bounded strings
+
+| ROS type                | Python type   |
+| ----------------------- | ------------- |
+| static array            | builtins.list |
+| unbounded dynamic array | builtins.list |
+| bounded dynamic array   | builtins.list |
+| bounded string          | builtins.str  |
+
+### Properties
+
+The class has same-named properties for every field of the message.
+The setter of a property will ensure that ROS type constraints beyond the Python type are enforced.
+
+### "Constants"
+
+A constant is defined as a class variable with uppercase name.
+
+### Constructors
+
+The *__init__* function initializes all members with their default value.
+
+The class constructor supports only keyword arguments for all members.
+The short reason for this is that if code would rely on positional arguments to construct data objects changing a message definition would break existing code in various ways.
+Since this would discourage evolution of message definitions the data structures should be populated by setting the members separately, e.g. using the setter methods.
+
+### Magic methods
+
+<div class="alert alert-warning" markdown="1">
+  <b>TODO:</b> decide which magic methods should be provided
+</div>
+
+## Services
+
+For a service a `class` with the same name is generated.
+
+The class contains only two `types`:
+
+- `Request` which is the type of the request part of the service
+- `Response` which is the type of the request part of the service
+
+The generated code is split across multiple files the same way as message are.
+
+### Request and response messages
+
+For the request as well as the response part messages are being generated which have the same name as the service with a suffix being either `_Request` or `_Response`.
```

Comment:
This sentence was awkward for me to read, how about: "The request and response components for a Service are messages whose names are made up of the Service's name and either a `_Request` or `_Response` suffix respectively."